tlslite/utils/python_tripledes.py

```diff
@@ -0,0 +1,491 @@
+#################################################
+#               Documentation                   #
+#################################################
+
+# Author:   Todd Whiteman
+# Date:     16th March, 2009
+# Verion:   2.0.0
+# License:  Public Domain - free to do as you wish
+# Homepage: http://twhiteman.netfirms.com/Des.html
+#
+# Modified by: Adam Varga, 2018
+#
+# A pure python implementation of the DES and Triple DES
+# encryption algorithms using CBC mode. Triple DES class is
+# implemented by utilising the DES base. Triple DES is
+# DES-EDE3 with a 24 byte key, or DES-EDE2 with a 16 byte key.
+
+"""
+Class initialization
+--------------------
+pyDes.Des(key, iv)
+pyDes.Python_TripleDES(key, iv)
+
+key -> Bytes containing the encryption key. 8 bytes for DES, 16 or 24 bytes
+       for Triple DES
+iv  -> Initialization Vector in bytes. Length must be 8 bytes.
+"""
+
+import sys
+import warnings
+
+# PY_VER is used to handle Python2 and Python3 differences.
+PY_VER = sys.version_info
+
+
+def new(key, iv):
+    """Operate this 3DES cipher."""
+
+    return Python_TripleDES(key, iv)
+
+
+class _baseDes(object):
+    """The base class shared by DES and triple DES."""
+
+    def __init__(self, iv):
+        self.block_size = 8
+        if iv:
+            if len(iv) != self.block_size:
+                raise ValueError("Invalid Initialization Vector (iv) must be"
+                                 " {0} bytes long".format(self.block_size))
+            iv = self._guard_against_unicode(iv)
+        else:
+            raise ValueError("Initialization Vector (iv) must be supplied")
+
+        self.iv = iv
+
+    def _guard_against_unicode(self, data):
+        """Check the data for valid datatype and return them.
+
+        Only accept byte strings or ascii unicode values.
+        Otherwise there is no way to correctly decode the data into bytes.
+        """
+
+        if PY_VER < (3, ):
+            if isinstance(data, unicode):
+                raise ValueError("Only bytes, bytearray or memoryview "
+                                 "objects of them should be passed, "
+                                 "not Unicode strings")
+        else:
+            if isinstance(data, str):
+                warnings.warn("Only bytes, bytearray or memoryview "
+                              "objects of them should be passed",
+                              DeprecationWarning,
+                              stacklevel=1)
```

Comment:
given that this emits errors like this:
```
/home/travis/build/tomato42/tlslite-ng/tlslite/utils/python_tripledes.py:74: DeprecationWarning: Only bytes, bytearray or memoryview objects of them should be passed
  stacklevel=1)
```
`1` here is not correct, will likely need `3` – we want the error to be emitted from user code, not this module's code